rpcs3/Emu/RSX/RSXThread.cpp

```diff
@@ -2449,21 +2458,24 @@ namespace rsx
 	//Pause/cont wrappers for FIFO ctrl. Never call this from rsx thread itself!
 	void thread::pause()
 	{
-		external_interrupt_lock.store(true);
-		while (!external_interrupt_ack.load())
+		while (UNLIKELY(external_interrupt_lock.exchange(true)))
 		{
-			if (Emu.IsStopped())
-				break;
+			std::this_thread::yield(); // TODO
```

Comment:
Don't fall back to heavyweight scheduler behavior when you only need to pause for a few microseconds.